api/envoy/config/cluster/v3/circuit_breaker.proto

```diff
@@ -97,6 +97,11 @@ message CircuitBreakers {
     // :ref:`Circuit Breaking <arch_overview_circuit_break_cluster_maximum_connection_pools>` for
     // more details.
     google.protobuf.UInt32Value max_connection_pools = 7;
+
+    // The maximum number of connections a given host can have. If not specified the default is
+    // unlimited. This is especially useful for dynamic forward proxy, where per cluster limits
+    // make less sense thatn per host limits.
+    google.protobuf.UInt32Value per_host_connection_limits = 9;
```

Comment:
Nit: I would use `max_connections_per_host` for terminology consistency with the rest of the circuit breakers.

Reply:
I still wonder whether instead of adding this here, we instead would have a `Thresholds per_host_threasholds` parallel to the cluster level thresholds. My concern is that we are going to end up with a lot of other per host limits eventually and we should just start with the entire message, even if we say that currently only connection limits are supported?

I don't feel super strongly about this but throwing it out there. @htuch thoughts?

/wait

Reply:
SGTM. Is the idea that these limits are not just circuit breakers then (and why we don't nest under circuit breakers)?

Reply:
Ok, I've added as per host threshholds with a note (and config guards) that only the connection limit is supported for now.  Also added tests and release notes so this should be good for full review.

Reply:
LGTM